libsweep/include/queue.h

```diff
@@ -0,0 +1,68 @@
+#ifndef SWEEP_QUEUE_62C8F42E8DD5_H
+#define SWEEP_QUEUE_62C8F42E8DD5_H
+
+/*
+ * Thread-safe queue.
+ * Implementation detail; not exported.
+ */
+
+#include <stdint.h>
+
+#include <atomic>
+#include <condition_variable>
+#include <mutex>
+#include <queue>
+#include <utility>
+
+namespace sweep {
+namespace queue {
+
+template <typename T> class queue {
+public:
+  queue(int32_t max) : max_size(max), the_queue(), the_mutex(), the_cond_var() {}
+
+  // Empty the queue
+  void flush() {
+    std::unique_lock<std::mutex> lock(the_mutex);
+    while (!the_queue.empty()) {
+      the_queue.pop();
+    }
```

Comment:
I was under the impression clear does not work with std::queue... or did you mean to use the method name `clear` as opposed to `flush`. I agree that is probably a better naming choice.